src/gui/render.cpp

```diff
@@ -441,19 +441,25 @@ static void RENDER_Reset( void ) {
 			height = MakeAspectTable( skip, render.src.height, yscale, yscale);
 		}
 	}
-/* Setup the scaler variables */
-	double par;  /* the pixel aspect ratio of the source pixel array */
-	par = (double)width / height / 4 * 3; /* MS-DOS screen is always 4:3 */
+
+	// Setup the scaler variables
 
 	if (dblh)
 		gfx_flags |= GFX_DBL_H;
 	if (dblw)
 		gfx_flags |= GFX_DBL_W;
 
-	#if C_OPENGL
-		GFX_SetShader(render.shader_src);
-	#endif
-	gfx_flags=GFX_SetSize(width,height,gfx_flags,gfx_scalew,gfx_scaleh,&RENDER_CallBack,par);
+#if C_OPENGL
+	GFX_SetShader(render.shader_src);
+#endif
+
+	// The pixel aspect ratio of the source image, assuming 4:3 screen
+	const double real_par = 3.0 / 4.0 * width / height;
```

Comment:
The PAR calculation is a bit tricky to read with two divisions ( `# / # / #` ), which the original line also had.
Suggest something like this, which is a bit more intuitive on first glance:
``` c++
const double real_par = (3.0 * width) / (4.0 * height);
```

Reply:
Good point; I'll go with a slightly different version, grouping respective widths and heights together.

```C++
const double real_par = (width / 4.0) / (height / 3.0);
```

Interesting tidbit: your version and both mine versions are numerically more stable than the original code was: for 320x200, in our versions the result is exactly 1.2 while old code was calculating 1.20000000000000022204. :)